src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -760,40 +765,60 @@ public boolean isKnownParticipant( UUID studyId, String participantId ) {
     private Set<UUID> getNotificationEnabledStudies(
             List<Map<FullQualifiedName, Set<Object>>> studies,
             SearchApi searchApi ) {
+        Set<UUID> result = new HashSet<>();
+
         Set<UUID> studyEntityKeyIds = studies
                 .stream()
                 .map( study -> UUID.fromString( study.get( ID_FQN ).iterator().next().toString() ) )
                 .collect( Collectors.toSet() );
 
-        Set<UUID> studyIds = studies
-                .stream()
-                .map( study -> UUID.fromString( study.get( STRING_ID_FQN ).iterator().next().toString() ) )
-                .collect( Collectors.toSet() );
+        //mapping from studyEKID -> studyId
+        Map<UUID, UUID> studyEntityKeyIdMap = new HashMap<>();
+
+        studies.forEach( study -> {
+            UUID studyId = UUID.fromString( study.get( STRING_ID_FQN ).iterator().next().toString() );
+            UUID entityKeyId = UUID.fromString( study.get( ID_FQN ).iterator().next().toString() );
+            studyEntityKeyIdMap.put( entityKeyId, studyId );
+        } );
```

Comment:
You're already looping over the studies above on line 770. You can merge / simplify this logic:

```
Map<UUID, UUID> idMap = studies
    .stream()
    .collect( Collectors.toMap(
        study -> UUID.fromString( study.get( ID_FQN ).iterator().next().toString() ),
        study -> UUID.fromString( study.get( STRING_ID_FQN ).iterator().next().toString() )
    ) );

Set<UUID> studyEntityKeyIds = idMap.keySet();
```

Reply:
Awesome.